### PR TITLE
feat(shield) Add Chalice support

### PR DESCRIPTION
**Pull request by joshajohnson**
Adding support for [Chalice](https://customkbd.com/products/chalice-pre-order).
This is my first ZMK PR so apologies for any issues and thanks in advance :)

## Board/Shield Check-list
 - [X] This board/shield is tested working on real hardware
 - [X] Definitions follow the general style of other shields/boards upstream ([Reference](https://zmk.dev/docs/development/new-shield))
 - [X] `.zmk.yml` metadata file added
 - [X] Proper Copyright + License headers added to applicable files (Generally, we stick to "The ZMK Contributors" for copyrights to help avoid churn when files get edited)
 - [X] General consistent formatting of DeviceTree files
 - [X] Keymaps do not use deprecated key defines (Check using the [upgrader tool](https://zmk.dev/docs/codes/keymap-upgrader))
 - [X] `&pro_micro` used in favor of `&pro_micro_d/a` if applicable
 - [X] If split, no name added for the right/peripheral half
 - [X] `.conf` file has optional extra features commented out

### CODE REVIEW

File: app/boards/shields/chalice/chalice.zmk.yml
Comment (Nicell) [inline app/boards/shields/chalice/chalice.zmk.yml:8]:
```suggestion
  - keys
  - underglow
```

**Review by Nicell — CHANGES_REQUESTED** [review]
Generally looks great, one small change, and then I was curious when this shield will start to ship, as we'd like to wait until then to merge it in.

**Comment by joshajohnson** [comment]
Thanks for the review @Nicell 😄

Chalice is currently shipping, so if it's possible to merge now that would be great.
We caught the split backspace issue in the field from one of the early units.

**Review by Nicell — APPROVED** [review]

**Pull request merged**
